shared/src/balancer/event_handler.rs

```diff
@@ -183,6 +177,43 @@ impl PoolRegistry {
             .unique_by(|pool| pool.pool_id)
             .collect()
     }
+}
+
+/// The BalancerPool struct represents in-memory storage of all deployed Balancer Pools
+#[derive(Derivative)]
+#[derivative(Debug)]
+pub struct DynPoolRegistry {
+    /// Used for O(1) access to all pool_ids for a given token
+    pools_by_token: HashMap<H160, HashSet<H256>>,
+    /// WeightedPool data for a given PoolId
+    pools: HashMap<H256, RegisteredWeightedPool>,
+    #[derivative(Debug = "ignore")]
+    data_fetcher: Box<dyn PoolDataFetching>,
+}
+
+impl DynPoolRegistry {
+    pub fn new(data_fetcher: Box<dyn PoolDataFetching>) -> Self {
+        DynPoolRegistry {
+            pools_by_token: Default::default(),
+            pools: Default::default(),
+            data_fetcher,
+        }
+    }
+
+    pub fn merge(&self, other: &DynPoolRegistry) -> PoolRegistry {
+        tracing::debug!("Merging Balancer Pool Registries");
```

Comment:
I included this log too see how often we are merging (since it requires cloning each registry to build) and I think the pool fetcher will request this every time... Could be a problem, but I haven't managed to come up with a better way of exposing the merged Registry to the pool fetcher on demand.

Reply:
One idea is too keep track of `last_updated`, but this value can only be derived from that of each registry as either the min or the max. Of course this value wont ever be up to date with the current block number so even if we give the pool fetcher info about last_updated, it would always try to build a new one.

Reply:
This isn't relevant anymore right?